internal/weapons/catalyst/mappa/mappa.go

```diff
@@ -40,46 +42,39 @@ func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile
 			return false
 		}
 
-		if !char.StatusIsActive(stackKey) {
-			stacks = 1
-			char.AddStatus(stackKey, stackDuration, true)
-			c.Log.NewEvent("mappa proc'd", glog.LogWeaponEvent, char.Index).
-				Write("stacks", stacks).
-				Write("expiry (without hitlag)", c.F+stackDuration)
-		} else if stacks < 2 {
-			stacks++
-			char.AddStatus(stackKey, stackDuration, true)
-			c.Log.NewEvent("mappa proc'd", glog.LogWeaponEvent, char.Index).
-				Write("stacks", stacks).
-				Write("expiry (without hitlag)", c.F+stackDuration)
+		if !char.StatusIsActive(buffKey) {
+			w.stacks = 0
+		}
+		if w.stacks < 2 {
+			w.stacks++
 		}
+
+		m := make([]float64, attributes.EndStatType)
+		m[attributes.PyroP] = dmg * float64(w.stacks)
+		m[attributes.HydroP] = dmg * float64(w.stacks)
+		m[attributes.CryoP] = dmg * float64(w.stacks)
+		m[attributes.ElectroP] = dmg * float64(w.stacks)
+		m[attributes.AnemoP] = dmg * float64(w.stacks)
+		m[attributes.GeoP] = dmg * float64(w.stacks)
+		m[attributes.DendroP] = dmg * float64(w.stacks)
```

Comment:
nit: cleaner to make list of attributes to set and loop over:
```go
for _, attr := range elementAttributes {
  m[attr] = dmg * float64(w.stacks)
}

Reply:
leaving as is (many other places do it like this and I don't feel like changing them in this PR)